src/SingleCellAnnotator.imjoy.html

```diff
@@ -406,56 +418,90 @@
     },
 
     async uploadAnnotation(){
-      // Default options are marked with *
-      const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
-      sendData.discarded = this.current_image_data.discarded;
-      if (this.current_image_data.discarded){
-        // delete sendData.mask;
-        sendData.mask = null;
+      // skip upload annotation if image is annotated and no new change
+      if (this.current_image_data.annotation_status&&(this.current_image_data.mask_adjusted===null)&&(this.current_image_data.location_adjusted===null)){
+        // skip uploadannotation
+        return
       }else{
-        sendData.bbox = this.current_image_data.mask_geojson.bbox;
-        sendData.location = {};
-        if (this.current_image_data.label_display_mode == 'single'){
-          sendData.location[this.current_image_data.location] = this.current_location_selected;
+        // Default options are marked with *
+        const sendData = JSON.parse(JSON.stringify(sendDataTemplate));
+        sendData.discarded = this.current_image_data.discarded;
+        if (this.current_image_data.discarded){
+          //delete sendData.mask;
+          sendData.mask = null;
+        }else{
+          sendData.bbox = this.current_image_data.mask_geojson.bbox;
+          sendData.location = {};
+          if (this.current_image_data.label_display_mode == 'single'){
+            // do not upload annotion if no location selected
+            if (this.current_location_selected===null){
+              if (this.current_image_data.mask_adjusted===null){
```

Comment:
```suggestion
              if (!this.current_image_data.mask_adjusted){
```

Reply:
Not correct. I have this.current_location_selected===false. which is different from null